sources/rest_api/__init__.py

```diff
@@ -254,21 +257,39 @@ def paginate_resource(
                 client: RESTClient = client,
                 incremental_object: Optional[Incremental[Any]] = incremental_object,
                 incremental_param: IncrementalParam = incremental_param,
+                row_filter: Any = row_filter,
+                transform: Any = transform,
+                exclude_columns: List[str] = exclude_columns,
             ) -> Generator[Any, None, None]:
+                def exclude_elements(item: Any, exclude_columns: List[str]) -> Any:
+                    for key in exclude_columns:
+                        del item[key]
+                    return item
+
                 if incremental_object:
                     params[incremental_param.start] = incremental_object.last_value
                     if incremental_param.end:
                         params[incremental_param.end] = incremental_object.end_value
 
-                yield from client.paginate(
+                for page in client.paginate(
```

Comment:
Is it guaranteed that `page` is an array and never a dict? Say the API returned `{...}` data in a top level dict and there was never an array in the response.